Replace OneGet assemblies with these generated by OneGet.org

jianyunt opened the pull request:
Removed PackageManagement source code
Removed existing PackageMangement test code and replace with 8 tests case as acceptance test.
Use Save-Module to pull down PackageMangement from myget.
Changed to alpha.11 in download.sh to enable use PowerShellGet and OneGet in the build

Fix 
Issue https://github.com/PowerShell/PowerShell/issues/1355 
Issue https://github.com/PowerShell/PowerShell/issues/2347

jianyunt committed:
Update build.psm1 to pull PackageManagement assemblies

jianyunt committed:
Use alpha.11 instead of alpha.12 to allow using OneGet to install modules from myget

jianyunt committed:
update project.json

jianyunt committed:
Add PackageManagement acceptance test cases

jianyunt committed:
remove PackageManagement source code

jianyunt committed:
remove packagemanagement test code

vors commented on build.psm1:
```diff
@@ -2508,6 +2517,95 @@ function Clear-PSRepo
+        }
+    }
+
+    log ("Name='{0}', Dsetination='{1}', Repository='{2}'" -f $Name, $Destination, $RepositoryName)
```
typo: `Dsetination`

vors commented on build.psm1:
```diff
@@ -2508,6 +2517,95 @@ function Clear-PSRepo
+    }
+
+    # pull down the module
+    log ("running save-module $Name")
```
You don't need `()` around `log` calls

vors commented on build.psm1:
```diff
@@ -2508,6 +2517,95 @@ function Clear-PSRepo
+    PowerShellGet\Save-Module @command -Force -Verbose:$VerbosePreference
+
+    # Clean up
+    if($needRegister)
```
that's cool 👍

vors reviewed (commented):
Overall, looks good to me

vors commented on build.psm1:
```diff
@@ -63,6 +63,7 @@ function Start-PSBuild {
         [switch]$ResGen,
         [switch]$TypeGen,
         [switch]$Clean,
+        [switch]$NoPSModuleRestore,
```
I would flip the default behavior and make it `-PSModuleRestore`.
Very tiny `Start-PSBuild` should be the bare minimum of things sufficient to run PowerShell.
It also should be incremental-friendly.

jianyunt replied:
changed to PSModuleRestore

jianyunt committed:
# This is a combination of 3 commits.
# The first commit's message is:

Changed to PSModuleRestore switch, i.e., by default no PSModule install

# This is the commit message #2:

install PowerShell modules to publish folder as well as one level up

# This is the commit message #3:

removed workaround

vors commented on build.psm1:
```diff
@@ -393,15 +393,21 @@ cmd.exe /C cd /d "$location" "&" "$($vcVarsPath)\vcvarsall.bat" "$NativeHostArch
+        $powerShellPath = Split-Path $publishPath -Parent
+        $destination = Join-Path -Path $powerShellPath -ChildPath "Modules"
+
+        if(Test-Path -path $destPublishPath)
```
This logic is not correct

1. On incremental builds, it would Silently not change the folder
1. When user provide a custom `binDir`, it will pollute parent directory.

Please, consider fix `Start-PSPester` to understand both `-Publish` and no-publish options.
It could be as simple as add `-Publish` switch.

vors replied:
Ok, lets merge the Fix in #2723 and we can proceed with your changes then.

Please revert back 969ac05d706d968eb534e886b5a5b9a4ce14a5bc

vors commented:
@jianyunt to avoid this in the future, please join Microsoft GitHub organization and make your membership public. 
![image](https://cloud.githubusercontent.com/assets/816680/20449318/7500fe58-ad9d-11e6-8a2e-e1458684ae5b.png)

bmanikm commented on build.psm1:
```diff
@@ -2508,6 +2528,95 @@ function Clear-PSRepo
+        if($regVar)
+        {
+            PowerShellGet\UnRegister-PSRepository -Name $RepositoryName
+        }
```
this is not needed as we are registering only if we there is no existing repo with the same source location.

jianyunt replied:
If the repository exists, $needRegister is false. This code won't get executed. 
Only the case when the same repo exists but is pointing to some other source location.

---
In reply to: [88761457](https://github.com/PowerShell/PowerShell/pull/2711#discussion_r88761457) [](ancestors = 88761457)

bmanikm commented on build.psm1:
```diff
@@ -2508,6 +2528,95 @@ function Clear-PSRepo
+        {
+            log "Unregistering PSRepository with name: $RepositoryName"
+            PowerShellGet\UnRegister-PSRepository -Name $RepositoryName
+        }
```
can be simplified with below
            log "Unregistering PSRepository with name: $RepositoryName"
            PowerShellGet\Get-PSRepository -Name $RepositoryName -ErrorAction SilentlyContinue | PowerShellGet\UnRegister-PSRepository -Name $RepositoryName

jianyunt replied:
sure. Will integrate it

---
In reply to: [88761662](https://github.com/PowerShell/PowerShell/pull/2711#discussion_r88761662) [](ancestors = 88761662)

bmanikm commented:
:shipit:

jianyunt committed:
add psmodulerestore in travis.ps1

vors commented on test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1:
```diff
@@ -62,6 +62,15 @@ $script:MyDocumentsScriptsPath = Microsoft.PowerShell.Management\Join-Path -Path
+# Check if the PackageManagement works in the base-oS or PowerShellCore
+$PSHome
+$PSVersionTable
+$env:PSModulePath
```
It's better to remove these lines. They will pollute the test output.

jianyunt committed:
Removed PowerShellGet source and replaced with ones from myget

vors commented on build.psm1:
```diff
@@ -407,6 +407,7 @@ cmd.exe /C cd /d "$location" "&" "$($vcVarsPath)\vcvarsall.bat" "$NativeHostArch
         $publishPath = Split-Path $Options.Output -Parent
         log "Restore PowerShell modules to $publishPath"
         Restore-PSModule -Name PackageManagement -Destination (Join-Path -Path $publishPath -ChildPath "Modules")
+        Restore-PSModule -Name PowerShellGet -Destination (Join-Path -Path $publishPath -ChildPath "Modules")
```
Can you, please, make `Name` parameter a `string[]` and pass an array of strings

vors replied:
Fixed it in 391a787

Commit:
Make Restore-PSModule -Name parameter a [string[]]

Commit:
Make Start-PSBootstrap on macOS better

No errors if brew dependencies already present
